src/main/java/stef/projects/console/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,170 @@
+package stef.projects.console.repository;
+
+import stef.projects.console.config.DatabaseConnection;
+import stef.projects.console.domain.User;
+import stef.projects.console.domain.UserRole;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserRepositoryImpl implements UserRepository {
+
+
+    private static final String SAVE_USER_QUERY = "insert into \"user\" values (default, ?, ?, ?, ?, ?, ?);";
+    private static final String DELETE_USER_QUERY = "delete from \"user\" where id = ?";
+    private static final String UPDATE_USER_QUERY = "update \"user\" set first_name = ?, last_name = ?, username = ?, email = ?, \"password\" = ?, \"user_role_id\" = ? where id = ?;";
+    private static final String SELECT_BY_ID_QUERY = "select * from \"user\" as u inner join \"user_role\" as r on u.\"user_role_id\" = r.id where u.id = ?;";
+    private static final String SELECT_ALL_QUERY = "select * from \"user\" as u inner join \"user_role\" as r on u.\"user_role_id\" = r.id ;";
+    private static final String SELECT_BY_EMAIL_QUERY = "select * from \"user\" as u inner join \"user_role\" as r on u.\"user_role_id\" = r.id where email = ?;";
+    private static final String SELECT_BY_USERNAME_AND_PASSWORD_QUERY = "select * from \"user\" as u inner join \"user_role\" as r on u.\"user_role_id\" = r.id where username = ? and \"password\" = ?;";
+    private static final String SELECT_BY_USER_ROLE_QUERY = "select * from \"user\" as u inner join \"user_role\" as r on u.\"user_role_id\" = r.id where \"value\" = ?;";
+    private static final String SELECT_BY_FIRST_OR_LAST_NAME_QUERY = "select * from \"user\" as u inner join \"user_role\" as r on u.\"user_role_id\" = r.id where first_name = ? or last_name = ?;";
+
+    @Override
+    public boolean saveUser(User user) throws SQLException {
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(SAVE_USER_QUERY);
+        fillPreparedStatement(user, statement, false);
+        int update = statement.executeUpdate();
+        return extractStatus(update);
+    }
+
+    @Override
+    public boolean updateUser(User user) throws SQLException {
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(UPDATE_USER_QUERY);
+        fillPreparedStatement(user, statement, true);
+        int update = statement.executeUpdate();
+        return extractStatus(update);
```

Comment:
You can do "return(extractStatus(statement.executeUpdate()));" since you are noy using it for anything else